### ISSUE TITLE
Refactor the Listing block to make components better split and reusable

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#3613]
I want to start a discussion on how to think about the listing block and its templates.

The current situation is: each variation controls the layout of the "item listings" and also how each "item" looks like in the listing. With the introduction of the Teaser item, we will have Teaser variation templates that we'll probably want to reuse as "Listing block result items". We also have many implementations of the Slideshow / Carousel concepts which should be independent of the way each item is displayed.

My proposal is to clearly separate the "listing layout" from the "item". For example, right now we have a slideshow implemented, the Image Galery. What if I want the slideshow behavior, but instead of Images I want to display Content cards or whatever. The slideshow implementation itself is not reusable as there's no separation between the Listing Layout and the Item.

**Comment by davisagli** [comment]
We need to figure out the specifics, but in general this sounds good to me. It should help separate the concern of "how this item is rendered" from the concern of "where this item is rendered", and make it easier to reuse a component between a block that displays a single item and a block that displays multiple items.

**Comment by erral** [comment]
It is an interesting idea indeed, but thinking a bit as an end user having so many different options perhaps makes it harder to use the block.

Lately we are adding a heading and a heading level selector, I thought that perhaps the No Results thingy could be also an option, and now we are thinking on differentiating 'Listing Variation' and 'Item Variation'.

It is indeed interesting and useful, and I am in favour, but I think that we need to take into account the average end-user too.

**Comment by erral** [comment]
As a follow up to previous coment, In a current project we have added a variation for the listing  which has a function to render each of the items, and that function decides according to the item to be rendered each time, which component to use to render that item.

It does not allow the end user
to select which 'item Variation' to use, but lets the developer to implement different conditions im code and then decide which component to render.

The deciding function could be registered in the component registry and thus let the developer customize it however she needs.

**Comment by tiberiuichim** [comment]
> which has a function to render each of the items, and that function decides according to the item to be rendered each time
Yes, that is another use case that I like. For example, in a card listing, the Event card could look different from a File card or a News item card, etc. But be aware that right now there's an edge case, maybe not important, the Listing block, when used as "folder contents (no query defined)" doesn't have access to the whole summarized items, complete with metadata. Your listing may not work the same as a folder content.

In my opinnion the Volto blocks picture can be made more complete by having a few extra tools at our disposal:

- block transformations - let's say I have a block Slider and one Carousel. I want be able to "mutate" their `@type` if their internal data is the same or can be adapted. This can go even further: imagine I select a bunch of blocks and I apply a "grouping" transformation, which can create a Column or Accordion to contain them.
- make variations available from the BlockChooser
- default block values, also for variations. If the Listing block has variations Slider and Carousel, and I pick the Carousel variation from the BlockChooser, it should automatically create a Listing block with the Carousel Variation selected.
- make the presentation part of the block relevant to the editors, rather then the source of data. As an editor, I can choose "Carousel" or "Slideshow" from the block chooser, then I can choose the source of data for the carousel: either a listing or manually adding the items (listing vs teaser grid).
